pkg/things/thing.go

```diff
@@ -175,15 +189,34 @@ func (t *Thing) RequestAccessToken(scopes ...string) (response AccessTokenRespon
 	if err != nil {
 		return
 	}
-	info, err := t.Client.AMInfo()
-	if err != nil {
-		return
-	}
-	requestBody, err := signedJWTBody(session, info.AccessTokenURL, info.ThingsVersion, NewGetAccessToken(scopes))
-	if err != nil {
-		return
+
+	var requestBody io.Reader
+	var format commandFormat
+	if session.HasRestrictedToken() {
+		info, err := t.Client.AMInfo()
+		if err != nil {
+			return response, err
+		}
+		jws, err := signedJWTBody(session, info.AccessTokenURL, info.ThingsVersion, NewGetAccessToken(scopes))
+		if err != nil {
+			return response, err
+		}
+		requestBody = strings.NewReader(jws)
+		format = commandFormatJOSE
+	} else {
+		b, err := json.Marshal(
+			struct {
+				getAccessTokenPayload
+				commandRequest
+			}{getAccessTokenPayload{Scope: scopes},
+				commandRequest{Token: session.token}})
```

Comment:
I think the token should only be added in the `coapclient` implementation. This will send it along with the HTTP request as well.